src/main/java/ProjektiPizza/demo/security/JwtAuthenticationProvider.java

```diff
@@ -0,0 +1,44 @@
+package ProjektiPizza.demo.security;
+
+import ProjektiPizza.demo.entity.User;
+import ProjektiPizza.demo.token.JwtAuthenticationToken;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.authentication.dao.AbstractUserDetailsAuthenticationProvider;
+import org.springframework.security.core.AuthenticationException;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.stereotype.Component;
+
+
+@Component
+public class JwtAuthenticationProvider extends AbstractUserDetailsAuthenticationProvider {
+
+    @Autowired
+    private JwtValidator validator;
+
+    @Override
+    protected void additionalAuthenticationChecks(UserDetails userDetails, UsernamePasswordAuthenticationToken usernamePasswordAuthenticationToken) throws AuthenticationException {
+    }
+
+    @Override
+    protected User retrieveUser(String username, UsernamePasswordAuthenticationToken usernamePasswordAuthenticationToken) throws AuthenticationException {
+        JwtAuthenticationToken jwtAuthenticationToken = (JwtAuthenticationToken) usernamePasswordAuthenticationToken;
+        String token = jwtAuthenticationToken.getToken();
+
+        User user = validator.validate(token);
+
+        if (user == null) {
+            throw new RuntimeException("JWT token is incorrect");
```

Comment:
This returns the response to the client as string, right? Maybe it would be better if we do this as a constructed JSON.

Reply:
i'll try to constructed JSON